test/functional/legacy/031_close_commands_spec.lua

```diff
@@ -0,0 +1,123 @@
+-- Test for commands that close windows and/or buffers
+-- :quit
+-- :close
+-- :hide
+-- :only
+-- :sall
+-- :all
+-- :ball
+-- :buf
+-- :edit
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('Commands that close windows and/or buffers', function()
+  setup(clear)
+
+  it('is working', function()
+    insert('testtext')
+
+    feed('GA 1<Esc>:$w! Xtest1<CR>')
+    feed('$r2:$w! Xtest2<CR>')
+    feed('$r3:$w! Xtest3<CR>')
+    execute('n! Xtest1 Xtest2')
+    feed('A 1<Esc>:set hidden<CR>')
+
+    -- Test for working :n when hidden set
+    -- Write "testtext 2"
+    execute('n')
+    execute('w! test.out')
```

Comment:
I think the expect can be split up. Looks like lines are always just appended to `test.out`. So for example, instead of this `execute`, an expect could be used:

```
expect('testtext2')
```